lstchain/reco/dl0_to_dl1.py

```diff
@@ -166,63 +183,84 @@ def r0_to_dl1(input_filename=get_dataset_path('gamma_test_large.simtel.gz'), out
-        add_prefix=True,
-        # overwrite=True
-    ) as writer:
+    ### Write extra information to the DL1 file
+    if is_simu:
+        event = next(iter(source))
+        write_array_info(event, output_filename)
+        write_mcheader(event.mcheader, output_filename, obs_id=event.r0.obs_id, filters=filters, metadata=metadata)
+        subarray = event.inst.subarray
+
+    with HDF5TableWriter(filename=output_filename,
+                         group_name='dl1/event',
+                         mode='a',
+                         filters=filters,
+                         add_prefix=True,
+                         # overwrite=True,
+                         ) as writer:
 
         print("USING FILTERS: ", writer._h5file.filters)
 
-        # build a mapping of tel_id back to tel_index:
-        # (note this should be part of SubarrayDescription)
-        idx = np.zeros(max(subarray.tel_indices) + 1)
-        for key, val in subarray.tel_indices.items():
-            idx[key] = val
-
-        # the final transform then needs the mapping and the number of telescopes
-        tel_list_transform = partial(utils.expand_tel_list,
-                                     max_tels=len(event.inst.subarray.tel) + 1,
-                                     )
-
-        writer.add_column_transform(
-            table_name='subarray/trigger',
-            col_name='tels_with_trigger',
-            transform=tel_list_transform
-        )
+        if is_simu:
+            # build a mapping of tel_id back to tel_index:
+            # (note this should be part of SubarrayDescription)
+            idx = np.zeros(max(subarray.tel_indices) + 1)
+            for key, val in subarray.tel_indices.items():
+                idx[key] = val
+
+            # the final transform then needs the mapping and the number of telescopes
+            tel_list_transform = partial(utils.expand_tel_list,
+                                         max_tels=len(event.inst.subarray.tel) + 1,
+                                         )
+
+            writer.add_column_transform(
+                table_name='subarray/trigger',
+                col_name='tels_with_trigger',
+                transform=tel_list_transform
+            )
 
+
+        ### EVENT LOOP ###
         for i, event in enumerate(source):
             if i % 100 == 0:
                 print(i)
 
-
             event.dl0.prefix = ''
             event.mc.prefix = 'mc'
             event.trig.prefix = ''
 
             # write sub tables
             write_subarray_tables(writer, event, metadata)
 
-            if not custom_calibration:
+            if not custom_calibration and is_simu:
                 cal(event)
 
+            if not is_simu:
+                r0_r1_calibrator.calibrate(event)
+                r1_dl1_calibrator(event)
+
+
             for ii, telescope_id in enumerate(event.r0.tels_with_data):
 
+                if not is_simu:
+                    combine_channels(event, telescope_id, 4095)
```

Comment:
Should not be 21 ? (Due to gain selection issue in presented in LST-reco call of 04/11/2019)

Reply:
The function `combine_channels` is working on R0 waveforms to get the channel mask so this should be correct.